src/adloader.js

```diff
@@ -107,6 +109,10 @@ export function loadExternalScript(url, moduleCode, callback, doc) {
 
     jptScript.src = tagSrc;
 
+    if (attributes) {
+      setScriptAttributes(jptScript, attributes);
```

Comment:
It is needed. Those attributes need to be set before the script is attached to the DOM. I'll investigate the cache key

Reply:
@dgirardi RE: caching attributes. If a script is cached, loadExternalScript will return that script immediately upon retrieving it from the cache. The setAttributes function is only ever called on the novel script generated inside the requestResource function, which is never reached when a resource has already been cached. So, to answer your question, multiple calls will always result in only the first script tag, and thus the first set of attributes, being returned. It would also be undesirable to include these attributes on the cache key, as they will already be present on the script tag and would both be unused and a violation of the single source of truth principle.

TLDR: Caching attributes would be redundant and the pr as written will never result in attributes being overwritten.

Reply:
I've been told in the past that I can be a bit blunt, which doesn't always come off the way I intend. So I want to flag that I am enjoying engaging in the topic of how this feature should be architected.

As it stands, if the same url is requested multiple times, Prebid basically stiff arms the second request and returns the original tag without modification. This includes dropping any callbacks from the second request. This could be considered a bug. 

For the purposes of the SpotX adapter, though, changing the attributes for a tag that already has attributes would not work. I'm not sure if it would break something or if it would just be ignored, but I do know it wouldn't work. So for the purposes for which this feature is being added, the current implementation is the desired one, if not necessarily "proper". 

I would also say that I think it is reasonable to say that a resource can only be loaded once, and any other requests for the same resource will simply go to the cache. Whether this is appropriate or whether it makes more sense to continually modify the request's callbacks or attributes is use case dependent. Given that the current architecture seems to not have raised any concerns so far, I would say that the simplest, and therefore probably best response to avoiding a hard to find bug would be adding a log warning that the resource has already been requested and won't be modified.

As for docs, this function is not present in Prebid's external facing api, and not currently documented. No documentation will be changed as a result of this pr.

Reply:
> As it stands, if the same url is requested multiple times, Prebid basically stiff arms the second request and returns the original tag without modification. This includes dropping any callbacks from the second request. 

I don't think the last bit is true (but I haven't tested), from reading the code I think the callback will run. I agree that if it doesn't, it looks like a bug to me - but I don't think it'd be in the scope of this PR (and after such a long time, it is now a feature :)

> For the purposes of the SpotX adapter, though, changing the attributes for a tag that already has attributes would not work.

But that's not what I'm proposing, which is creating a new script element (because the cache is either missing or bypassed). Basically if the attributes are different, treat it as a different resource. It's similar to what the implementation does with the `doc` argument, which was recently added.

> As for docs, this function is not present in Prebid's external facing api, and not currently documented. 

What I mean is, note it in the param documentation, so that if someone gets bit by this I can blame them instead of you :) I saw that you did that while I was typing this, that's good enough for me.